### DISCUSSION TITLE
Comments on first use

### DISCUSSION THREAD

**Discussion by eeholmes** [nmfs-opensci/nmfs-jhub#2] [General]
from Kathryn

Positives
- It's great the Rstudio is an option for the interface, but there are other options available, too!
- I was easily able to open an R session and start coding in a script. The UX was just like being in an R studio session locally!
- The git pane in the Jhub interface before you navigate over to R studio seems pretty intuitive, I was able to clone a repo quickly
- I was pleasantly surprised that devtools::document() just worked (from opening an R session in the terminal in the jhub interface)! I see usethis is also available, so all set up for R pkg development (which I know is just one use case)

Points of confusion
- After cloning the repo in the Jhub interface, I opened up R studio and seemed to be in a different directory? The repo I cloned was nowhere to be found, even after looking in the root (/) directory and copy/pasting over the file path from the jhub interface. Does R studio open in a different place or something?  ( I did realize I could make a new project in R studio and clone there, that did work and seemed to persist)
- Even though I logged in with GitHub, my git and github credentials weren't automatically stored. I was prompted for my name and email address to set up the git config, and then my email and password (which I thought wasn't even allowed by github anymore, I thought you had to use a PAT?) to push to github. I'm not sure if there is a behind the scenes way to integrate git and github better once you log in, given I've already put in my github credentials? If not, instructions could probably fix this issue.

**Comment by eeholmes** [comment]
@k-doering-NOAA Re having Git remember you

Follow these instructions https://snowex-2022.hackweek.io/preliminary/git.html

Basically

1. Open up a terminal
```
git config --global user.name "Attendee Name"
git config --global user.email "attendee@hackweek.com"
git config --global pull.rebase false
git config --global credential.helper store
```

2. Make a PAT on GitHub

3. Head back to the JupyterHub and do a push. It'll ask for your username and password. After that it will remember.

**Comment by eeholmes** [comment]
@k-doering-NOAA Re the cloned repo being nowhere to be found in RStudio. Yeah I think these use different user folders. You'll need to clone the repo into RStudio. Note if moving between RStudio and JupyterLab, the potential for merge conflicts is high. Best to choose one or the other for working on code. Note you can work on R code in Jupyter Notebooks I think and via the terminal as you discovered.

**Comment by k-doering-NOAA** [comment]
Ok, thanks! I'll try following the instructions when I have time to try it out again. Having these things documented somewhere for the nmfs-jhub instance would be helpful.
